src/main/java/uk/gov/pay/products/persistence/dao/ProductMetadataDao.java

```diff
@@ -33,4 +34,15 @@ public ProductMetadataDao(Provider<EntityManager> entityManager) {
                 .setParameter("productExternalId", productExternalId)
                 .getResultList();
     }
+
+    public Optional<ProductMetadataEntity> findByProductsExternalIdAndKey(String productExternalId, String key) {
+        String query = "SELECT metadata FROM ProductMetadataEntity metadata, ProductEntity product " +
+                "WHERE metadata.productEntity = product AND product.externalId = :productExternalId " +
```

Comment:
I'm not 100% familiar with how this would work in JPA land but do we need to specify the join condition in the WHERE if there is a foreign key?